src/test/java/jp/co/yahoo/yosegi/reader/TestYosegiReader.java

```diff
@@ -0,0 +1,307 @@
+    ByteArrayInputStream in = new ByteArrayInputStream( blocks );
+    YosegiReader reader = new YosegiReader();
+    AndExpressionNode index = new AndExpressionNode();
+    index.addChildNode(
+        new ExecuterNode( new StringExtractNode( "column2" )
+        , new PerfectMatchStringFilter( "p" ) )
+    );
+    Configuration readerConfig = new Configuration();
+    readerConfig.set( "spread.reader.read.column.names" , "[[\"column2\"]]" );
+    reader.setBlockSkipIndex( index );
+    // Skip Block-3
+    reader.setNewStream( in , blocks.length , readerConfig );
+    assertEquals( reader.hasNext() , true );
+
+    // Block-1 Spread-1
+    List<ColumnBinary> raw = reader.nextRaw();
+    assertEquals( reader.getCurrentSpreadSize().intValue() , 4 );
+    assertEquals( raw.size() , 0 );
+    assertEquals( reader.hasNext() , true );
+
+    // Block-1 Spread-2
+    raw = reader.nextRaw();
+    assertEquals( reader.getCurrentSpreadSize().intValue() , 4 );
+    assertEquals( raw.size() , 0 );
+    assertEquals( reader.hasNext() , true );
+
+    // Block-2 Spread-1
+    raw = reader.nextRaw();
+    assertEquals( reader.getCurrentSpreadSize().intValue() , 4 );
+    assertEquals( raw.size() , 0 );
+    assertEquals( reader.hasNext() , true );
+
+    // Block-2 Spread-2
+    raw = reader.nextRaw();
+    assertEquals( reader.getCurrentSpreadSize().intValue() , 4 );
+    assertEquals( raw.size() , 0 );
+    assertEquals( reader.hasNext() , false );
+  }
+
+  @Test
+  public void T_EmptyPushdownAndBlockRead_1() throws IOException {
+    byte[] blocks = createTestBinary();
+    ByteArrayInputStream in = new ByteArrayInputStream( blocks );
+    YosegiReader reader = new YosegiReader();
+    AndExpressionNode index = new AndExpressionNode();
+    index.addChildNode(
+        new ExecuterNode( new StringExtractNode( "column" )
+        , new PerfectMatchStringFilter( "p" ) )
+    );
+    Configuration readerConfig = new Configuration();
+    // Skip Block-1, Block-2
+    reader.setBlockSkipIndex( index );
+    // Read Block-1, Block-2
+    reader.setNewStream( in , blocks.length , readerConfig , 0 , blockSize * 2 );
+    assertEquals( reader.hasNext() , false );
+  }
+
+  @Test
+  public void T_EmptyPushdownAndBlockRead_2() throws IOException {
+    byte[] blocks = createTestBinary();
+    ByteArrayInputStream in = new ByteArrayInputStream( blocks );
+    YosegiReader reader = new YosegiReader();
+    AndExpressionNode index = new AndExpressionNode();
+    index.addChildNode(
+        new ExecuterNode( new StringExtractNode( "column" )
+        , new PerfectMatchStringFilter( "p" ) )
+    );
+    Configuration readerConfig = new Configuration();
+    // Skip Block-1, Block-2
+    reader.setBlockSkipIndex( index );
+    // Read Block-2
+    reader.setNewStream( in , blocks.length , readerConfig , blockSize , blockSize );
+    assertEquals( reader.hasNext() , false );
+  }
+
+  @Test
+  public void T_EmptyPushdownAndBlockRead_3() throws IOException {
```

Comment:
please describe what the code wants to test in the method name instead of number extension.